web/client/epics/login.js

```diff
@@ -42,6 +43,7 @@ const refreshTokenEpic = (action$, store) =>
 const reloadMapConfig = (action$, store) =>
     action$.ofType(LOGIN_SUCCESS, LOGOUT)
     .filter(() => pathnameSelector(store.getState()).indexOf("viewer") !== -1)
+    .filter(() => hasMapAccessLoadingError(store.getState))
```

Comment:
What happens on logout?  
Try to reload the resource metadata (canEdit, canView) for anonymous user. If the permission are view only it's ok (Check that you are not able to save anyway).
If you can not see the map anymore, you could redirect the user to the home page or show the same error.